packages/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -299,9 +318,6 @@ export const GridRootStyles = styled('div', {
       display: 'flex',
       alignItems: 'center',
     },
-    [`& .${c['columnHeader--last']}`]: {
-      overflow: 'hidden',
-    },
```

Comment:
I understand this was added here https://github.com/mui/mui-x/pull/12808, though I don't fully understand what problem it solves.

The cutting off of the separator looks bad, so keen to find another solution if the problem still exists.

<img width="136" alt="Screenshot 2024-08-23 at 13 29 20" src="https://github.com/user-attachments/assets/c524b8ca-c9ad-4138-823e-b35a7dceb296">

Reply:
It's necessary to avoid this issue: 

https://github.com/user-attachments/assets/96e8b78d-f880-437a-8313-5d9b3280997b

Perhaps we can hide overflow conditionally when there's horizontal scrolling?

Reply:
Makes sense - made that change here: https://github.com/mui/mui-x/pull/14293/commits/14345bb90cd62f0b0eaf013ad8997c522a3d49fa